provider/pkg/await/condition/observer.go

```diff
@@ -84,6 +84,12 @@ func (oo *ObjectObserver) Observe(e watch.Event) error {
 	oo.mu.Lock()
 	defer oo.mu.Unlock()
 	obj, _ := e.Object.(*unstructured.Unstructured)
+
+	// Do nothing if this is a stale object.
+	if obj.GetGeneration() < oo.obj.GetGeneration() {
+		return nil
+	}
```

Comment:
Nit: Would you agree that a more precise comparison would be of the resourceVersion than the generation? You're trying to observe versions that are later than what was returned from the mutating create/update. Your logic would needlessly re-observe the initial version, right?

In other words, **generation** is a concept of versioned specs, but the awaiter logic is primarily about **status changes** (where generation doesn't change).